flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/crd/status/FlinkDeploymentStatus.java

```diff
@@ -53,4 +53,7 @@ public class FlinkDeploymentStatus extends CommonStatus<FlinkDeploymentSpec> {
 
     /** Information about the TaskManagers for the scale subresource. */
     private TaskManagerInfo taskManager;
+
+    /** The jobId, if it was set by the operator. */
+    private String overrideJobId;
```

Comment:
I think we should simply set it into the jobStatus.jobId and avoid introducing an extra field for it. 

As long as the job is running we should assume that the jobStatus reflects the correct jobid

Reply:
All the current logic uses the jobStatus.jobId for savepointing, cancelling, observing etc, it would avoid a lot of complications :)

Reply:
Reusing jobStatus.jobId is not perfect as we are using the same field for two different purposes. But it works pretty well and is better than adding another field to status for something that is going to be phased out when >=1.16 becomes the minimum version. PTAL